### PR TITLE
feat: Service exceptions to period tickets

### PR DESCRIPTION
**Pull request by NitishAroraKPMG**
## Description

For the period ticket, service exception needs to be displayed

## Testing instructions

Create a period ticket with zeo zone

### CODE REVIEW

File: repos/fdbt-site/src/components/UserDataUploads.tsx [hunk 1]
```diff
@@ -25,133 +25,264 @@ const UserDataUploadComponent = ({
     poundsOrPence,
     csrfToken,
     backHref,
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/components/UserDataUploads.tsx:27]: Why are we editing the UserDataUploads component? The change is only required to happen on the csvZoneUpload page

File: repos/fdbt-site/src/pages/csvZoneUpload.tsx [hunk 2]
```diff
@@ -37,11 +56,67 @@ export const isFareZoneAttributeWithErrors = (
...
 ): fareZoneAttribute is FareZoneWithErrors => (fareZoneAttribute as FareZoneWithErrors).errors !== undefined;
 
-export const getServerSideProps = (ctx: NextPageContextWithSession): { props: UserDataUploadsProps } => {
+export const getServerSideProps = async (ctx: NextPageContextWithSession): Promise<{ props: UserDataUploadsProps }> => {
     const fareZoneAttribute = getSessionAttribute(ctx.req, FARE_ZONE_ATTRIBUTE);
     const errors: ErrorInfo[] =
         fareZoneAttribute && isFareZoneAttributeWithErrors(fareZoneAttribute) ? fareZoneAttribute.errors : [];
 
+    const nocCode = getAndValidateNoc(ctx);
+    //updateSessionAttribute(ctx.req, SERVICE_LIST_EXEMPTION_ATTRIBUTE, undefined);
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/csvZoneUpload.tsx:65]: commented out code

File: repos/fdbt-site/src/pages/csvZoneUpload.tsx [hunk 3]
```diff
@@ -37,11 +56,67 @@ export const isFareZoneAttributeWithErrors = (
...
+            : [];
+    let chosenDataSourceServices;
+    if (modesAttribute) {
+        chosenDataSourceServices = await getTndsServicesByNocAndModes(nocCode, modesAttribute.modes);
+    }
+    chosenDataSourceServices = await getServicesByNocCodeAndDataSource(nocCode, dataSourceAttribute.source);
+
+    const selectedLineIds = selectedServices.map((service) => service.lineId);
+    //console.log(chosenDataSourceServices);
+    //console.log({ selectedServices });
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/csvZoneUpload.tsx:112]: more commented out code

File: repos/fdbt-site/src/pages/csvZoneUpload.tsx [hunk 4]
```diff
@@ -54,10 +129,18 @@ export const getServerSideProps = (ctx: NextPageContextWithSession): { props: Us
             csvTemplateDisplayName: 'Download fare zone CSV template - File Type CSV - File Size 673B',
             csvTemplateAttachmentUrl: FareZoneExampleCsv,
             csvTemplateSize: '673B',
-            errors,
+            errors: [...errors],
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/csvZoneUpload.tsx:132]: Why have we changed this?

File: repos/fdbt-site/src/pages/csvZoneUpload.tsx [hunk 5]
```diff
@@ -54,10 +129,18 @@ export const getServerSideProps = (ctx: NextPageContextWithSession): { props: Us
...
             detailSummary: "My fare zone won't upload",
             csrfToken: getCsrfToken(ctx),
             backHref: '',
+            showPriceOption: true,
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/csvZoneUpload.tsx:136]: Why are we showing something to do with prices? This is a zone upload, not a fare triangle upload.

**Comment by alainajaffer** [comment]
Add exemptions to confirmation section

File: repos/fdbt-site/src/pages/csvZoneUpload.tsx (hunk 3 above)
Comment (NitishAroraKPMG) [reply repos/fdbt-site/src/pages/csvZoneUpload.tsx]: removed

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**